core/object.cpp

```diff
@@ -1942,15 +1943,15 @@ Object::~Object() {
 
 	const StringName *S = NULL;
 
+	if (_emitting) {
+		//@todo this may need to actually reach the debugger prioritarily somehow because it may crash before
+		ERR_PRINTS("Object was freed or unreferenced while a signal is being emitted from it. Try connecting to the signal using 'CONNECT_DEFERRED' flag, or use queue_free() to free the object (if this object is a Node) to avoid this error and potential crashes.");
```

Comment:
Can we get details identifying the object like `get_string()` or is this too late in the destructor?

Reply:
I think we can, I'll give it a try

Reply:
MMhhh... `get_class_name` seems to always return "Object" (probably because the child destructor has been called already, and `get_class_name` is an override function).
I can still get the `instance_id` which might be useful I guess.

Reply:
I typoed it initially, but I meant you can use `this->to_string()` for that. See https://github.com/godotengine/godot/pull/35088/files/f38bfccf42ccf16fd84cb8df41a19a2c3a0f4923#diff-21145ad4eb02c796f1e240892c937e41L1522-L1524

Reply:
I typoed it initially but I meant that you can use `this->to_string()` for that. See https://github.com/godotengine/godot/pull/35088/files/f38bfccf42ccf16fd84cb8df41a19a2c3a0f4923#diff-21145ad4eb02c796f1e240892c937e41L1522-L1524

Reply:
oh, nice! Well, I've added that. But at least in my case is still returning `Object` (though it also adds the ID automatically so that's the right way to print anyway).